### PR TITLE
Add Laravel 10 Compatibility

### PR DESCRIPTION
**Pull request by snakindia**
Fixed issue with workflow build file

### CODE REVIEW

**Comment by joskfg** [comment]
The build is not being triggered. Please fix it.

File: src/database/seeds/ScrapedDatasetSeeder.php [hunk 1]
```diff
@@ -11,11 +11,11 @@ class ScrapedDatasetSeeder extends Seeder
      */
     public function run()
     {
-        factory(ScrapedDataset::class, 2)->create();
+        ScrapedDataset::factory(2)->create();
```
Comment (joskfg) [inline src/database/seeds/ScrapedDatasetSeeder.php]: Could you please be consistent and use `factory()->count(2)` instead of `factory(2)` ? Check other places just in case.

File: tests/Unit/Scraper/Application/ConfiguratorTest.php [hunk 2]
```diff
@@ -209,11 +212,10 @@ public function whenTryToFindNewXpathButNotFoundItShouldLogItAndResetVariant():
 
         Log::shouldReceive('warning')
             ->with("Field ':field-2:' with value ':value-2:' not found for ':scrape-url:'.");
-
-        $this->expectsEvents(ConfigurationScraped::class);
-
+            
         try {
             $this->configurator->configureFromDataset($posts);
+            Event::assertDispatched(ConfigurationScraped::class);
```
Comment (joskfg) [inline tests/Unit/Scraper/Application/ConfiguratorTest.php:218]: This is dead code. Review other tests with the same behavior please.

File: tests/Unit/Scraper/Listeners/ConfigureScraperTest.php [hunk 3]
```diff
@@ -209,13 +209,14 @@ public function whenTheIdStoreIsNotAvailableItShouldThrowAnUnexpectedValueExcept
         Log::shouldReceive('debug');
         Log::shouldReceive('error')
             ->with("Error scraping ':scrape-url:'", ['message' => ':error:']);
-        $this->expectsEvents(ScrapeFailed::class);
 
+
```
Comment (joskfg) [inline tests/Unit/Scraper/Listeners/ConfigureScraperTest.php]: Avoid double empty lines and Space the Event::assertDispatched to let the code breathe there, check other tests. I see that I need to add an improved checkstyle :pray:.

File: tests/Unit/Scraper/Listeners/ScrapeTest.php [hunk 4]
```diff
@@ -51,15 +54,15 @@ public function whenConfigurationDoesNotExistItShouldThrowAnEvent(): void
...
-        $this->expectsEvents(InvalidConfiguration::class);
-
+            
         $scrape = new Scrape(
             $this->config,
             $this->xpathFinder,
             Log::getFacadeRoot()
         );
-
+
```
Comment (joskfg) [inline tests/Unit/Scraper/Listeners/ScrapeTest.php]: Same about check style. Avoid not needed spaces.

File: tests/Unit/Scraper/Listeners/ConfigureScraperTest.php [hunk 5]
```diff
@@ -112,18 +115,15 @@ public function whenIsCalculatedItShouldReturnExtractedDataAndStoreTheNewConfig(
...
+        Event::assertDispatched(Scraped::class);
 
         /** @var Scraped $event */
-        $event = collect($this->firedEvents)->filter(function ($event): bool {
-            $class = Scraped::class;
-
-            return $event instanceof $class;
-        })->first();
-        self::assertEquals(
+        $firedEvents = collect(Event::dispatched(Scraped::class));
```
Comment (joskfg) [inline tests/Unit/Scraper/Listeners/ConfigureScraperTest.php]: Why collect is needed here or in the other tests?

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php [hunk 6]
```diff
@@ -101,7 +99,7 @@ public function whenDatasetDoesNotExistAndTheDatasetsLimitHasNotBeenReachedItSho
     public function whenDatasetDoesNotExistAndTheDatasetsLimitHasReachedItShouldDeleteTheExcess(): void
     {
         $type = ':type:';
-        factory(ScrapedDataset::class, UpdateDataset::DATASET_AMOUNT_LIMIT + 10)->create([
```
Comment (joskfg) [inline tests/Unit/Scraper/Listeners/UpdateDatasetTest.php:104]: Now this test is not testing what is described in the method name.

**Review by joskfg — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -17,6 +17,12 @@ jobs:
     - name: Validate composer.json and composer.lock
       run: composer validate --strict
 
+    - name: Set up PHP
```
Comment (joskfg) [inline .github/workflows/build.yml:20]: Thanks a lot! :clap:

File: src/database/seeds/ScrapedDatasetSeeder.php (hunk 1 above)
Comment (snakindia) [reply src/database/seeds/ScrapedDatasetSeeder.php]: This is done

File: tests/Unit/Scraper/Listeners/ScrapeTest.php (hunk 4 above)
Comment (snakindia) [reply tests/Unit/Scraper/Listeners/ScrapeTest.php]: Done

File: tests/Unit/Scraper/Listeners/ConfigureScraperTest.php (hunk 3 above)
Comment (snakindia) [reply tests/Unit/Scraper/Listeners/ConfigureScraperTest.php]: Done

File: src/database/seeds/ScrapedDatasetSeeder.php (hunk 1 above)
Comment (snakindia) [reply src/database/seeds/ScrapedDatasetSeeder.php]: Done

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php (hunk 6 above)
Comment (snakindia) [reply tests/Unit/Scraper/Listeners/UpdateDatasetTest.php:104]: The changes are reverted back and working fine.

File: tests/Unit/Scraper/Listeners/ConfigureScraperTest.php (hunk 5 above)
Comment (snakindia) [reply tests/Unit/Scraper/Listeners/ConfigureScraperTest.php]: It is not required because we have no need to return the array of all event's instances. Removed it.

File: tests/Unit/Scraper/Application/ConfiguratorTest.php (hunk 2 above)
Comment (snakindia) [reply tests/Unit/Scraper/Application/ConfiguratorTest.php:218]: We have tested it that when an event is to be triggered then it returns true.

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php [hunk 8]
```diff
@@ -16,15 +15,13 @@ class UpdateDatasetTest extends TestCase
...
+    private UpdateDatasets $updateDataset;
 
     public function setUp(): void
     {
         parent::setUp();
 
-        Log::spy();
-
-        $this->updateDataset = new UpdateDataset();
+        $this->UpdateDatasets = new UpdateDataset();
```
Comment (xaviapa) [inline tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: Why did you change the name of the property? This one is not defined. Moreover, it should be in camelCase.

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php [hunk 9]
```diff
@@ -16,15 +15,13 @@ class UpdateDatasetTest extends TestCase
 {
     use DatabaseMigrations;
 
-    private UpdateDataset $updateDataset;
+    private UpdateDatasets $updateDataset;
```
Comment (xaviapa) [inline tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: Why renaming this class? UpdateDatasets does not exist.

Comment (snakindia) [reply tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: This was done because we were not able to access the public functions of factory class and were only getting the protected functions of the class which was initialized in the constructor of factory class inside vendor library. 

There was some conflict with the property name and some other in built library variable. When we changed the name then we were able to access the create function of factory class and then things started working.

Comment (xaviapa) [reply tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: You renamed `UpdateDataset` to `UpdateDatasets`, you are using a class that does not exist now, how can this work? We only have  `Joskfg\LaravelIntelligentScraper\Scraper\Listeners\UpdateDataset`.

Sorry, but I don’t get which private methods about a factory class you are talking about. `UpdateDataset` is a listener.

File: tests/Unit/Scraper/Application/ConfiguratorTest.php [hunk 10]
```diff
@@ -215,10 +215,11 @@ public function whenTryToFindNewXpathButNotFoundItShouldLogItAndResetVariant():
             
         try {
             $this->configurator->configureFromDataset($posts);
-            Event::assertDispatched(ConfigurationScraped::class);
         } catch (ConfigurationException $e) {
             self::assertEquals('Field(s) ":field-1:" not found.', $e->getMessage());
         }
+
+        Event::assertDispatched(ConfigurationScraped::class);
```
Comment (joskfg) [inline tests/Unit/Scraper/Application/ConfiguratorTest.php]: Almost! Please follow the example adding the assertions in the catch:
```
try {
   // Execution
   // Fail assertion
}
catch( <Exception> ) {
   // Asserts
}
```

And it can be even improved adding the fail assertion if it not launch the Exception.
```suggestion
            $this->fail('A ConfigurationException should be thrown')
        } catch (ConfigurationException $e) {
            Event::assertDispatched(ConfigurationScraped::class);
            self::assertEquals('Field(s) ":field-1:" not found.', $e->getMessage());
        }
```

The same should be applied for the other tests with the same schema.

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php [hunk 11]
```diff
@@ -16,14 +15,12 @@ class UpdateDatasetTest extends TestCase
 {
     use DatabaseMigrations;
 
-    private UpdateDataset $updateDataset;
+    private updateDataset $updateDataset;
```
Comment (xaviapa) [inline tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: The class name should start with uppercase. I was only talking about the property name.

Comment (joskfg) [reply tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: Please follow PSR-12

File: tests/Unit/Scraper/Listeners/UpdateDatasetTest.php [hunk 12]
```diff
@@ -101,11 +100,11 @@ public function whenDatasetDoesNotExistAndTheDatasetsLimitHasNotBeenReachedItSho
     public function whenDatasetDoesNotExistAndTheDatasetsLimitHasReachedItShouldDeleteTheExcess(): void
     {
         $type = ':type:';
-        factory(ScrapedDataset::class, UpdateDataset::DATASET_AMOUNT_LIMIT + 10)->create([
+        $test = ScrapedDataset::factory()->count(UpdateDataset::DATASET_AMOUNT_LIMIT + 10)->create([
```
Comment (joskfg) [inline tests/Unit/Scraper/Listeners/UpdateDatasetTest.php]: I think this `$test` variable is not used.

**Review by joskfg — APPROVED** [review]
